src/darwin/Framework/CHIP/MTRDevice_Concrete.mm

```diff
@@ -1178,6 +1182,10 @@ - (void)_scheduleSubscriptionPoolWork:(dispatch_block_t)workBlock inNanoseconds:
         MTRAsyncWorkItem * workItem = [[MTRAsyncWorkItem alloc] initWithQueue:self.queue];
         [workItem setReadyHandler:^(id _Nonnull context, NSInteger retryCount, MTRAsyncWorkCompletionBlock _Nonnull completion) {
             mtr_strongify(self);
+            if (!self) {
```

Comment:
`VerifyOrReturn(self);` ? Would be nice to have a common pattern for this that's used everywhere.

Reply:
Good idea - I've made this change.